dig/default.go

```diff
@@ -20,30 +20,30 @@
 
 package dig
 
-// Default graph is used for all the top-level inject calls
-var defaultGraph = newGraph()
+// DefaultGraph is used for all the top-level inject calls
+var DefaultGraph Graph = newGraph()
```

Comment:
The example I gave won't work.
need to be something like
``` golang
type resolveFunc func(interface{}) error
func New(info auth.CreateAuthInfo, middleware ...OutboundMiddleware) *http.Client {
     return new(info, dig.Resolve, middleware...)
}

func new(info auth.CreateAuthInfo, resolveFn resolveFunc, middleware ...OutboundMiddleware) *http.Client {...}
```

I hope dig can expose a `Resolver` interface or only return copy